### ISSUE TITLE
Common Voice posts to Basket missing email

### ISSUE THREAD

**Issue by pmac** [mozilla/voice-web#2048]
Hi @Gregoor. Starting again on May 13th I'm seeing a lot of messages from CV coming through without an email address. The data I see looks like this for example:

```python
data = {
    'created_at': '2019-04-29T22:24:39Z', 
    'days_interval': 1, 
    'first_contribution_date': '2018-02-07T18:38:21Z',
    'goal_reached_at': '2019-04-29T22:31:15Z',
}
```

**Comment by Gregoor** [comment]
Huh, thanks for the report! Looking into it now.

**Comment by Gregoor** [comment]
That's strange, I found a user (with an email) for that data and our logs also show that we've included that data in the sync.

@pmac do you have a date for me when that happened and/or how often it happened? I'm trying to find a pattern here

**Comment by pmac** [comment]
I have data from April 27 to now. From April 27 to May 12 there were no errors like this. Since May 13 there have been just over 4000 requests to the common voice endpoint with the proper api key but without email in the data. On May 24 there was a spike of 872. Most days were a lot lower than that, but there have been a decent number every day since the 13th.

**Comment by pmac** [comment]
I don't. I only have the data that was present when the error occurred. I have more examples though. It looks like it happens a lot, so I don't think it's an individual user with a data problem though. I could maybe get you access to Sentry for basket so you could poke around in the errors.

**Comment by pmac** [comment]
I think I've finally figured this out. It was my fault :( These tasks that were being called without emails were due to how the task was written in basket. The tasks would fail due to an error in SFDC, but the task had already modified the data passed to it by removing the email address and doing other things with it. But then when the error happened the task was retried by our queue, but the data passed to the retry was the modified version and had no email. I've filed https://github.com/mozmeao/basket/pull/205 to fix it.
